Widgets/BaseClass.aspx.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+
+namespace Widgets {
+    public partial class BaseClass : System.Web.UI.Page {
+        protected override void OnPreInit(EventArgs e) {
+            var master = Master as Site;
+            if (master != null && !string.IsNullOrEmpty(master.Theme)) {
+                Theme = master.Theme;
+            }
+
+            base.OnPreInit(e);
+        }
+    }
```

Comment:
When adding a base page class to your project, just add a C# class to your project instead of using the code behind class for an ASPX page.